### ISSUE TITLE
[Suggestion] 3D/OBJ/FBX model export of entities

### ISSUE THREAD

**Issue by ItsCrusade** [tsunamayo/Starship-EVO#2338]
I'd like to suggest Starship EVO to have a 3d model exporting feature.

I've spent countless hours rendering player-made ships from Starmade - as Starmade has a 3D Export feature. 

I'm not entirely sure of the technical limitations of such a feature, or what it would take to implement it, but I believe it would be a nice thing to have. You could produce high-quality renders of your ships/hovercrafts and any other creations in external programs, 3D Print your ships, import them into wallpaper engine, share them on Sketchfab and embed a 3d model instead of a static image into forums/Discord. 

Having such a feature would be a fantastic option in my option, as well as allow the game to be used as a utility - increasing the potential userbase.

Attached are some renders of Starmade creations. 
Made with Blender, rendered with Cycles.

![clean](https://user-images.githubusercontent.com/64144395/83792286-3166e980-a6de-11ea-96ef-ec5cf99b30dd.png)
![rend 06](https://user-images.githubusercontent.com/64144395/83792302-3b88e800-a6de-11ea-815c-48f4a13d2c31.png)
![render Solaris 4](https://user-images.githubusercontent.com/64144395/83792330-43e12300-a6de-11ea-810e-312af8d83938.png)
![render Solaris 5](https://user-images.githubusercontent.com/64144395/83792362-4e9bb800-a6de-11ea-8f13-90378eac1d81.png)

**Comment by ItsCrusade** [comment]
More details:

I've managed to pull most of a SEVO ship with some DX11 intercepts.

Just the more accurate lighting looks amazing - even without textures.

![progress3](https://user-images.githubusercontent.com/64144395/83814597-0a221380-a702-11ea-9042-c7264defbce0.png)
![progress4](https://user-images.githubusercontent.com/64144395/83814675-11e1b800-a702-11ea-93cb-59491b4bcd81.png)
![progress5](https://user-images.githubusercontent.com/64144395/83814680-13ab7b80-a702-11ea-8d5f-166d626fbbcc.png)

**Comment by tsunamayo** [comment]
Yes it looks cool on paper, the problem is that my rendering tech is hugely specific:
I have the stretch tech that is working on the GPU, and then the border tech, some block like Hull reading their neighbor information from a voxel structure on the GPU.
So I cant really just export an OBJ and expect to get the same textured result as in the game.

**Comment by ItsCrusade** [comment]
I'm not sure the capabilities of Unity - but is it possible to grab an entity after GPU transform functions are done? Kind of like what I've done through brute force.

**Comment by tsunamayo** [comment]
yeah now I dont think we can do that natively, and anyway like I told you the problem is not the mesh (that is possible, just time consuming as I would have to duplicate code on the CPU side), that's the textures. You would need a very custom shader in whatever application you are using.

**Comment by ItsCrusade** [comment]
Yeah that's totally fair!
The images of Starmade above are using a shader node tree I built in blender, and I went ahead and created a basic material node for my shuttle in Starship EVO.

Original: 
![Starship_EVO_2020-04-27_18-04-48](https://user-images.githubusercontent.com/64144395/83875264-219ce300-a77a-11ea-8fcb-cd6fc01b8b35.png)

Render
![SEVO](https://user-images.githubusercontent.com/64144395/83875274-282b5a80-a77a-11ea-8cc1-39c08d0ac4c0.png)

Thanks for the info Tsuna! :)
